### DISCUSSION TITLE
Which draw2d version still support Java 11?

### DISCUSSION THREAD

**Discussion by soerendomroes** [eclipse-gef/gef-classic#626] [Q&A] [answered]
Reading the release documentation, I see that with [gef 3.16 Java 17 is required](https://github.com/eclipse-gef/gef-classic/releases/tag/R_3_16).

However, trying to install an earlier version, I get the following error message:

```
org.eclipse.draw2d 3.14.100.202311222042 requires Execution Environment that matches (&(osgi.ee=JavaSE)(version=17)) but the current resolution context uses [a.jre.javase 11.0.0]
```

Is there a version that still supports Java 11 that is compatible with Eclipse platform 3.29.

**Comment by ptziegler** [comment, marked as answer]
> org.eclipse.draw2d 3.14.100.202311222042

The version of the GEF release doesn't necessarily need to match the plugin version. Draw2D 3.14.100 belongs to GEF [3.18.0](https://download.eclipse.org/justj/?file=tools/gef/classic/release/3.18.0/plugins) and therefore requires Java 17, as stated.

The latest release that still supports Java 11 is [3.15.0](https://download.eclipse.org/tools/gef/classic/release/3.15.0/).

> Eclipse platform 3.29.

Where did you get that number from? The latest 3.x release was 3.8.2. If you mean the 4.29 then this is again the 3.15.0, or for Draw2D specifically the 3.12.0.
